lib/Runtime/Types/PathTypeHandler.h

```diff
@@ -190,20 +245,38 @@ namespace Js
         BOOL ConvertToSharedNonExtensibleTypeIfNeededAndCallOperation(DynamicObject* instance, const PropertyRecord* operationInternalPropertyId, FType operation);
 
         template <bool isObjectLiteral>
-        DynamicType* PromoteType(DynamicType* type, const PropertyRecord* propertyId, bool shareType, ScriptContext* scriptContext, DynamicObject* object = nullptr, PropertyIndex* propertyIndex = nullptr);
+        DynamicType* PromoteType(DynamicType* type, const PathTypeSuccessorKey key, bool shareType, ScriptContext* scriptContext, DynamicObject* object = nullptr, PropertyIndex* propertyIndex = nullptr);
+        ObjectSlotAttributes * UpdateAttributes(Recycler * recycler, ObjectSlotAttributes * oldAttributes, uint8 oldPathSize, TypePath * newTypePath);
+        SetterSlotIndex * UpdateSetterSlots(Recycler * recycler, SetterSlotIndex * oldSetters, uint8 oldPathSize, TypePath * newTypePath);
```

Comment:
>SetterSlotIndex [](start = 8, length = 15)

Maybe explicit naming this PathTypeHandlerSettingSlotIndex since this is only for PathTypeHandlers?